File: split/printthedigs.pas
procedure printthedigs(k: eightbits);
begin
  while k > 0 do
  begin
    k := k - 1;
    if dig[k] < 10 then
      printchar(48 + dig[k])
    else
      printchar(55 + dig[k]);
  end;
end;

File: cpp/src/printthedigs.cpp
#include "printthedigs.h"
#include "printchar.h"

void printthedigs(eightbits k)
{
	while (k > 0)
	{
		k--;
		if (dig[k] < 10)
			printchar('0'+dig[k]);
		else
			printchar('A'+dig[k]-10);
	}
}
